aten/src/ATen/mps/MPSDevice.mm

```diff
@@ -22,10 +22,17 @@
 
 MPSDevice::MPSDevice(): _mtl_device(nil) {
   // Check that MacOS 12.3+ version of MPS framework is available
-  id mpsCD = NSClassFromString(@"MPSGraphCompilationDescriptor");
-  if (![mpsCD instancesRespondToSelector:@selector(optimizationLevel)]) {
-    // According to https://developer.apple.com/documentation/metalperformanceshadersgraph/mpsgraphcompilationdescriptor/3922624-optimizationlevel
-    // this means we are running on older MacOS
+  // Create the MPSGraph and check method introduced in 12.3+
+  // which is used by MPS backend.
+  id mpsCD = NSClassFromString(@"MPSGraph");
+  if (mpsCD == nil || ([mpsCD instancesRespondToSelector:@selector(LSTMWithSourceTensor:
```

Comment:
Nit: one does not need an explicit `nil` check here, as `objc_msgSend` returns `NO` for empty objects
```suggestion
  if ([mpsCD instancesRespondToSelector:@selector(LSTMWithSourceTensor:
```